src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/SymbolIsBannedInAnalyzersTests.cs

```diff
@@ -258,5 +261,54 @@ End Class
                 }
             }.RunAsync();
         }
+
+        [Fact]
+        public async Task UseBannedApi_ISourceGenerator()
+        {
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.Default.WithPackages(ImmutableArray.Create(new PackageIdentity("Microsoft.CodeAnalysis.Common", "4.4.0"))),
+                TestCode = """
+                    using Microsoft.CodeAnalysis;
+                    
+                    [Generator]
+                    class MyGenerator : ISourceGenerator
+                    {
+                        public void Initialize(GeneratorInitializationContext context)
+                        {
+                            context
+                                .RegisterForPostInitialization(
+                                    _ => { });
+                        }
+                        public void Execute(GeneratorExecutionContext context)
+                        {
+                            context
+                                .AddSource(
+                                    "Generated.cs",
+                                    "// <auto-generated/>");
+                        }
+                    }
+                    """,
+                TestState =
+                {
+                    AnalyzerConfigFiles =
+                    {
+                        ("/.editorconfig", """
+                        root = true
+
+                        [*]
+                        build_property.EnforceExtendedAnalyzerRules = true
+                        """),
+                    },
+                },
+                ExpectedDiagnostics =
+                {
+                    // /0/Test0.cs(8,9): error RS1035: The symbol 'GeneratorInitializationContext' is banned for use by analyzers: Non-incremental source generators should not be used, implement IIncrementalGenerator instead
+                    VerifyCS.Diagnostic("RS1035").WithSpan(8, 9, 10, 26).WithArguments("GeneratorInitializationContext", ": Non-incremental source generators should not be used, implement IIncrementalGenerator instead"),
+                    // /0/Test0.cs(14,9): error RS1035: The symbol 'GeneratorExecutionContext' is banned for use by analyzers: Non-incremental source generators should not be used, implement IIncrementalGenerator instead
+                    VerifyCS.Diagnostic("RS1035").WithSpan(14, 9, 17, 40).WithArguments("GeneratorExecutionContext", ": Non-incremental source generators should not be used, implement IIncrementalGenerator instead"),
```

Comment:
💡 The `WithLocation` API makes it easier to see where the diagnostic is appearing relative to source code.

```suggestion
                    VerifyCS.Diagnostic("RS1035").WithLocation(0).WithArguments("GeneratorInitializationContext", ": Non-incremental source generators should not be used, implement IIncrementalGenerator instead"),
                    // /0/Test0.cs(14,9): error RS1035: The symbol 'GeneratorExecutionContext' is banned for use by analyzers: Non-incremental source generators should not be used, implement IIncrementalGenerator instead
                    VerifyCS.Diagnostic("RS1035").WithLocation(1).WithArguments("GeneratorExecutionContext", ": Non-incremental source generators should not be used, implement IIncrementalGenerator instead"),
```